.kokoro/build-with-run.sh

```diff
@@ -32,6 +32,44 @@ popd
 export GOOGLE_CLOUD_PROJECT=nodejs-docs-samples-tests
 pushd github/nodejs-docs-samples/${PROJECT}
 
+# Verify changes worth testing.
+SIGNIFICANT_CHANGES="$(git --no-pager diff --name-only main..HEAD | grep -Ev '(\.md$|^\.github)' || true)"
+
+# If this is a PR with only insignificant changes, don't run any tests.
+if [[ -n ${KOKORO_GITHUB_PULL_REQUEST_NUMBER:-} ]] && [[ -z "$SIGNIFICANT_CHANGES" ]]; then
+  echo "No big changes. Not running any tests."
+  exit 0
+fi
+
+# CHANGED_DIRS is the list of directories that changed.
+# CHANGED_DIRS will be empty when run on main.
+CHANGED_DIRS=$(echo "$SIGNIFICANT_CHANGES" | tr ' ' '\n' | xargs dirname 2>/dev/null || true)
+
+# Default to not running the test.
+match=0
+
+# If CHANGED_DIRS is empty, default to running the tests.
+# This ensures nightly tests will run.
```

Comment:
**Praise:** It's hard to write self-documenting bash. 😅  So I highly appreciate your use of comments in this script!